imagenet/main.py

```diff
@@ -356,6 +362,9 @@ def run_validate(loader, base_progress=0):
                 i = base_progress + i
                 if args.gpu is not None and torch.cuda.is_available():
                     images = images.cuda(args.gpu, non_blocking=True)
+                if torch.xpu.is_available():
```

Comment:
I just imagine when someone will create a build with both cuda and xpu support and run it on the system with both cards :)

IMHO, I would add `elif` for both xpu and mps cases.